go-controller/hybrid-overlay/pkg/controller/node_linux.go

```diff
@@ -353,8 +353,156 @@ func getIPAsHexString(ip net.IP) string {
 	return asHex
 }
 
+// swaps out the new values of drIP and drMAC
+func (n *NodeController) recalculateFlowCache(oldDRIP net.IP, oldDRMAC net.HardwareAddr) {
+	n.flowMutex.Lock()
+	defer n.flowMutex.Unlock()
+
+	var ipReplace, macReplace int
+
+	if oldDRIP != nil {
+		ipReplace = -1
+	}
+	if oldDRMAC != nil {
+		macReplace = -1
+	}
+	oldDRMACRaw := strings.Replace(oldDRMAC.String(), ":", "", -1)
+	newDRMACRaw := strings.Replace(n.drMAC.String(), ":", "", -1)
+
+	portIPRawOld := getIPAsHexString(oldDRIP)
+	portIPRawNew := getIPAsHexString(n.drIP)
+
+	for _, entry := range n.flowCache {
+		for i, flow := range entry.flows {
+			replacementFlow := flow
+			replacementFlow = strings.Replace(replacementFlow, oldDRIP.String(), n.drIP.String(), ipReplace)
+			replacementFlow = strings.Replace(replacementFlow, portIPRawOld, portIPRawNew, ipReplace)
+
+			replacementFlow = strings.Replace(replacementFlow, oldDRMACRaw, newDRMACRaw, macReplace)
+			replacementFlow = strings.Replace(replacementFlow, oldDRMAC.String(), n.drMAC.String(), macReplace)
+			entry.flows[i] = replacementFlow
+
+		}
+	}
+}
+
+func (n *NodeController) createOrReplaceRoutes(mgmtPortLink netlink.Link, oldDRIP net.IP) error {
+	// hybridOverlay has been initialized and will needs to delete the old routes
+	if oldDRIP != nil {
+		if len(config.HybridOverlay.ClusterSubnets) > 0 {
+			for _, clusterEntry := range config.HybridOverlay.ClusterSubnets {
+				route := &netlink.Route{
```

Comment:
@JacobTanenbaum since this block is the same as loop below for all the nodes, can you pull it out into a function that just takes the `Dst` subnet?